ibis/expr/operations.py

```diff
@@ -2183,51 +2091,40 @@ class IdenticalTo(Comparison):
     pass
 
 
-class Between(BooleanValueOp):
-
-    input_type = [
-        rules.value(),
-        rules.value(name='lower_bound'),
-        rules.value(name='upper_bound')
-    ]
+class Between(ValueOp, BooleanValueOp):
```

Comment:
Why does this need to subclass from both `ValueOp` *and* `BooleanValueOp`?

Reply:
This PR might not reflect, but I've tried refactor non-that-aggressively :)